lib/dry/types/hash/schema.rb

```diff
@@ -4,15 +4,15 @@ class Hash < Definition
       class Schema < Hash
         attr_reader :member_types
 
-        def initialize(primitive, options = {})
+        def initialize(_primitive, options)
           @member_types = options.fetch(:member_types)
           super
         end
 
         def call(hash, meth = :call)
           member_types.each_with_object({}) do |(key, type), result|
             if hash.key?(key)
-              result[key] = type.__send__(meth, hash[key])
+              result[key] = type.public_send(meth, hash.fetch(key))
```

Comment:
I like that it conveys to me that we are certain of the key being present. I agree it is weird to have under `key?` but this is an intermediary step (I did this PR so I could refactor some of the duplication in the schemas). Can we leave it for now?